_posts/yyyy-mm-dd-ansible-czy-da-sie-uporzadkowac-chaos.md

```diff
@@ -0,0 +1,301 @@
+ok: [lan] => (item=14454)
+
+TASK [microservice-x : stop microservice-x] *********************************************************************************************************************************************************************
+
+TASK [microservice-x : start/restart microservice-x] ************************************************************************************************************************************************************
+changed: [lan]
+
+TASK [microservice-x : check if microservice-x process exists] **************************************************************************************************************************************************
+ok: [lan]
+
+TASK [microservice-x : report status of service] ************************************************************************************************************************************************************
+skipping: [lan]
+
+TASK [microservice-x : report status of service] ************************************************************************************************************************************************************
+ok: [lan] => {
+    "msg": "Microservice-x works"
+}
+
+PLAY RECAP **********************************************************************************************************************************************************************************************
+lan             : ok=12   changed=3    unreachable=0    failed=0    skipped=5    rescued=0    ignored=0 
+```
+Sposób prezentowania informacji przez Ansible jest dość czytelny. Instalację rozpoczyna uruchomienie playbooka o nazwie deploy microservice-x. Następnie wykonywany jest domyślny, ansible’owy task gathering_facts, który zbiera informację o hostach, na których wykonywane będą zmiany. Później już wykonywane są wszystkie zadania zdefiniowane w playbooku i wykorzystywanych rolach. Każde zadanie kończy się pewnym statusem. Na powyższym przykładzie widać, że zadania uzyskały takie statusy, jak: **ok**, **changed** czy **skipping**. Dzięki wspomnianej na początku idempotentności niektóre zadania zostały pominięte, ponieważ system był już w pożądanym stanie. Tym system przypisał status **ok**. Taski, które wymagały wykonania jakiejś zmiany na serwerze oznaczone zostały statusem **changed**. Status **skipping** został przypisany zadaniom, których wykonanie zostało oparte o pewien warunek. Jeżeli warunek nie był spełniony, to zadanie zakończyło się statusem **skipping**. 
+
+Każde wykonanie playbooka kończy się podsumowaniem. Podsumowanie zawiera zbiorczą informację dotyczącą wszystkich zadań oraz ich końcowych statusach. Gdyby okazało się, że jeden z kluczowych tasków nie może zostać wykonany, np. z powodu braku połączenia z serwerem docelowym, to skrypt zakończyłby swoje wykonanie na tym zadaniu, np.:
```

Comment:
Tu można uniknąć powtórzenia "podsumowaniem" i "podsumowanie" poprzez połączenie tych zdań

Reply:
to stary plik :)